sympy/solvers/solvers.py

```diff
@@ -382,77 +382,107 @@ def solve(f, *symbols, **flags):
         - piecewise combinations of the above
         - systems of linear and polynomial equations
         - systems containing relational expressions
+        - systems implied by undetermined coefficients
 
     Examples
     ========
 
-    The output varies according to the input and can be seen by example:
+    The default output varies according to the input and might
+    be a list (possibly empty), a dictionary, a list of
+    dictionaries or tuples, or an expression involving relationals.
+    An document describing the conditions under which different forms
+    of output are given is in explanations section of the documentation.
```

Comment:
```suggestion
    More about the conditions under which different forms
    of output can be found in these [explanations](../../doc/source/explanation/solve_output.rst).
```

Reply:
There is an anchor at the top of the file so it should be possible to do something like
```
:ref:`solve_output`
```
Then the link still works even if the file gets moved.

Reply:
I don't know understand how things are being looked up and from where. I'm not even sure which "this file" has the anchor. Does the change I made look right?

Reply:
Here's an example of an anchor and references:
```console
$ git grep domainsintro
doc/src/modules/polys/domainsintro.rst:.. _polys-domainsintro:
doc/src/modules/polys/domainsref.rst::ref:`polys-domainsintro`. This page lists the reference docs for the
doc/src/modules/polys/domainsref.rst::ref:`polys-domainsintro` for more of an explanation). There are three types
doc/src/modules/polys/domainsref.rst::py:class:`~.Domain` in the domain system (see :ref:`polys-domainsintro`).
doc/src/modules/polys/domainsref.rst::py:class:`~.Domain` in the domain system (see :ref:`polys-domainsintro`).
doc/src/modules/polys/index.rst:    domainsintro.rst
doc/src/modules/polys/numberfields.rst:This is explained in more detail :ref:`here<polys-domainsintro>`.
sympy/polys/domains/algebraicfield.py:    :ref:`polys-domainsintro`).
sympy/polys/domains/domain.py:    See :ref:`polys-domainsintro` for an introductory explanation of the
sympy/polys/domains/finitefield.py:    :ref:`polys-domainsintro`).
sympy/polys/domains/gaussiandomains.py:    :ref:`polys-domainsintro`).
sympy/polys/domains/gaussiandomains.py:    :ref:`polys-domainsintro`).
```